server/service/microsoft_mdm.go

```diff
@@ -579,3 +580,19 @@ func (svc *Service) GetAuthorizedSoapFault(ctx context.Context, eType string, or
 
 	return &soapFault
 }
+
+func (svc *Service) SignMDMMicrosoftClientCSR(ctx context.Context, subject string, csr *x509.CertificateRequest) ([]byte, string, error) {
+	// TODO: check if this method should require explicit authorization
+	svc.authz.SkipAuthorization(ctx)
+
+	cert, fpHex, err := svc.wstepCertManager.SignClientCSR(ctx, subject, csr)
+	if err != nil {
+		return nil, "signing wstep client csr", ctxerr.Wrap(ctx, err)
+	}
+
+	// TODO: if desired, the signature of this method can be modified to accept a device UUID so
```

Comment:
Yes, that's what is going to happen here. The subject to pass is going to be the device ID sent by the windows Enrollment client